src/Sulu/Bundle/TestBundle/Testing/KernelTestCase.php

```diff
@@ -0,0 +1,190 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\TestBundle\Testing;
+
+use Doctrine\Common\DataFixtures\Executor\ORMExecutor;
+use Doctrine\Common\DataFixtures\ProxyReferenceRepository;
+use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Doctrine\ORM\EntityManager;
+use InvalidArgumentException;
+use Sulu\Bundle\ContentBundle\Document\HomeDocument;
+use Sulu\Bundle\TestBundle\Kernel\SuluTestKernel;
+use Sulu\Component\Content\Document\WorkflowStage;
+use Symfony\Bundle\FrameworkBundle\Client;
+use Symfony\Component\Security\Core\Tests\Authentication\Token\TestUser;
+
+/**
+ * Base test case for functional tests in Sulu.
+ */
+abstract class KernelTestCase extends \PHPUnit_Framework_TestCase
+{
+    /**
+     * @var Container
+     */
+    private $container;
+
+    /**
+     * @var KernelInterface[]
+     */
+    private $kernelStack;
+
+    /**
+     * Return the test container.
+     *
+     * This container will use the default configuration of the kernel.
+     *
+     * If you require the container with a different kernel configuration
+     * you should create a new Kernel with the `getKernel` method and
```

Comment:
Kernel for a different environment.. different configuration implies different Container configs.